### PR TITLE
Bump steep and rbs

### PR DESCRIPTION
**Pull request by ybiquitous**
### Summary or Background

This change:

- updates the `steep` and `rbs` gems
- migrates from `git submodule` to `rbs collection`
- fixes or ignores new type errors

See also:

- [`rbs collection` doc](https://github.com/ruby/rbs/blob/d35ea838d0d0f7de7c71b4beaaac6666bb5f9805/docs/collection.md)
- [rbs collection was released! - DEV Community](https://dev.to/pocke/rbs-collection-was-released-4nmm)

### Issues or Pull Requests

None.

### CODE REVIEW

File: lib/runners/processor/actionlint.rb [hunk 1]
```diff
@@ -57,10 +57,11 @@ def analyzer_options
 
     def analyze(changes)
       stdout, stderr, status = capture3(analyzer_bin, *analyzer_options)
+      exitstatus = status.exitstatus
 
       # @see https://github.com/rhysd/actionlint/blob/v1.6.3/docs/usage.md#exit-status
       case
-      when [0, 1].include?(status.exitstatus)
+      when exitstatus && [0, 1].include?(exitstatus)
```
Comment (ybiquitous) [inline lib/runners/processor/actionlint.rb:64]: [note] [`Process::Status#exitstatus`](https://docs.ruby-lang.org/en/2.7.0/Process/Status.html#method-i-exitstatus) can return `nil` if the process is still running (mostly an edge case).

File: rbs_collection.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Download sources
```
Comment (ybiquitous) [inline rbs_collection.yaml:1]: [note] This file is generated by the `bundle exec rbs collection init` command.

File: lib/runners/processor/rubocop.rb [hunk 3]
```diff
@@ -72,12 +72,13 @@ def analyze(_changes)
       )
 
       _, stderr, status = capture3(cmd.bin, *cmd.args)
+      exitstatus = status.exitstatus
```
Comment (ybiquitous) [inline lib/runners/processor/rubocop.rb:75]: > __[RuboCop]__ Assignment Branch Condition size for analyze is too high. [<13, 58, 9> 60.12/50] ([view](https://sider.review/gh/repos/202486306/pulls/3088#issue-6915399))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `Metrics/AbcSize` | `convention` |
>
> References:
> - <https://docs.rubocop.org/rubocop/cops_metrics.html#metricsabcsize>
> - <http://c2.com/cgi/wiki?AbcMetric>
> - <https://en.wikipedia.org/wiki/ABC_Software_Metric>
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/3088#issue-6915399) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

Ignorable.

**Review by radomirbosak — APPROVED** [review]
Looks good! Thank you :+1:

**Pull request merged**